README.md

```diff
@@ -1,3 +1,227 @@
+
+   ```
+   As a library customer
+   I want to be able to search for books with a specific title, author, genre or ISBN
+   So I can find books I am interested in
+   ```
+
+## Project setup
+
+### Environment setup
+
+1. Docker - run
+   ```
+   docker run --name postgres -p 5432:5432 -e POSTGRES_PASSWORD=password -d postgres
+   ```
+2. pgAdmin - setup database serviers 'postgres'
+3. Postman
+
+### API setup
+
+1. change to working directory, i.e. /projects
+2. run
+   ```
+   git clone https://github.com/DavidCheungTW/book-library.git
+   cd book-library
+   npm install
+   ```
+3. Environment variables
+
+   Create a `.env` file for postman test:
+
+   ```
+   PGPASSWORD=password
+   PGDATABASE=book_library_dev
+   PGUSER=postgres
+   PGHOST=localhost
+   PGPORT=5432
+   PORT=3000
+   ```
+
+   Create a `.env.test` file for `npm test`:
+
+   ```
+   PGPASSWORD=password
+   PGDATABASE=book_library_dev_test
+   PGUSER=postgres
+   PGHOST=localhost
+   PGPORT=5432
+   PORT=3000
+   ```
+
+## Tests
+
+### Test by node
+
+1. use `book_library_dev_test` database
+2. change to book-library directory
+3. run `npm test` to execute the test
+4. add test case to \*.test.js files
+5. run `npm test` to execute the test again
+
+### Test by Postman
+
+1. use `book_library_dev` database
+2. change to book-library directory
+3. run `npm start` to start testing
+4. input test cases in Postman and check results (remark: suggest to use \*.test.js test cases first, then you can add your own test case)
+5. Example:
+
+   Add Author record:
+
+   POST : http://localhost:3000/authors
+
+   BODY/RAW/JSON :
+   `{"author":"J. K. Bowling"}`
+
+   Response Body:
+
+   ```
+   {
+   "id": 1,
+   "author": "J. K. Rowling",
+   "createdAt": "2023-02-27T09:32:56.215Z",
+   "updatedAt": "2023-02-27T09:32:56.215Z"
+   }
+   ```
+
+   Add Genre record:
+
+   POST : http://localhost:3000/genres
+
+   BODY/RAW/JSON :`{"genre":"Fiction"}`
+
+   Response Body:
+
+   ```
+   {
+   "id": 1,
+   "genre": "Fiction",
+   "updatedAt": "2023-02-27T09:51:31.388Z",
+   "createdAt": "2023-02-27T09:51:31.388Z"
+   }
+   ```
+
+   Add Book record:
+
+   POST : http://localhost:3000/books
+
+   BODY/RAW/JSON :`{"title":"harry potter and the deathly hallows","ISBN":"9780329499983","AuthorId":1,"GenreId":1}`
+
+   Response Body:
+
+   ```
+   {
+   "id": 2,
+   "title": "harry potter and the deathly hallows",
+   "ISBN": "9780329499983",
+   "AuthorId": 1,
+   "GenreId": 1,
+   "updatedAt": "2023-02-27T09:50:49.470Z",
+   "createdAt": "2023-02-27T09:50:49.470Z"
+   }
+   ```
+
+   Add Reader record:
+
+   POST : http://localhost:3000/readers
+
+   BODY/RAW/JSON :
+   `{"name":"David Cheung","email":"davidcheung@gmail.com","password":"12345678"}`
+
+   Response Body:
+
+   ```
+   {
+   "id": 1,
+   "name": "David Cheung",
+   "email": "davidcheung@gmail.com",
+   "updatedAt": "2023-02-27T09:40:11.047Z",
+   "createdAt": "2023-02-27T09:40:11.047Z"
+   }
+   ```
+
```

Comment:
It's good that you show the POST request routes and req.body needed to get the database started. It would also be good to say what constraints there are (ie password must be longer than 8 characters) and what datatypes are expected.
I would also like to see all the routes available for GET, PATCH and DELETE requests and only way to know is to look in the code. With these you should explain any req.params or req.body used